parser/src/parser/grammar.lalrpop

```diff
@@ -118,6 +119,24 @@ PeriodicColumn: PeriodicColumn = {
         PeriodicColumn::new(name, values),
 }
 
+// RANDOM VALUES
+// ================================================================================================
+
+RandomValues: RandomValues = {
+    "random_values" ":" <rand_values: RandValues> => rand_values
+}
+
+RandValues: RandomValues = {
+    <name: Identifier> ":" <index: Index> => RandomValues::new(name, index, vec![]),
+    <name: Identifier> ":" "[" "]" => RandomValues::new(name, 0, vec![]),
+    <name: Identifier> ":" "[" <rand_vec: CommaElems<RandElem>> "]" => RandomValues::new(name, rand_vec.iter().map(|v| v.size()).sum::<usize>(), rand_vec)
+}
+
+RandElem: RandBinding = {
+    <name: Identifier> => RandBinding::new(name, 1),
+    <name: Identifier> <index: Index> => RandBinding::new(name, index)
```

Comment:
We should probably use `Size` instead of index here, to be consistent with the `TraceCols` and the `PublicInput` declarations